grails-app/services/org/pih/warehouse/shipping/ShipmentService.groovy

```diff
@@ -1505,6 +1505,8 @@ class ShipmentService {
             debitTransaction.requisition = shipmentInstance.requisition
             debitTransaction.transactionNumber = identifierService.generateTransactionIdentifier()
 
+            debitTransaction.order = shipmentInstance.orders.findResult {Order it -> it.isReturnOrder() ? it : null}
```

Comment:
@jmiranda this is the part that I want to specifically revisit and improve on Monday. I want to check and set the order only in case we have the return order. 

> Why wouldn't we just return any order if there's an order on the shipment

Because we can have multiple orders in a case when it is not a Return Order, hence `any` might be problematic.

> What happens if we have multiple return orders on the shipment

Currently, Return Orders have one Order and one Shipment tied together, there is currently no option to do it differently (with multiple orders per shipment). Because of that, IMHO we can only set the order on the transaction only in case of return orders. I wanted to do it only when we have a case with Return Order. And I wanted to use for it the `getReturnOrder` transient (this one is already added in recently pushed PR made by Dariusz which I still have to review and merge). This would basically be:
```
if (shipmentInstance.returnOrder) {
    debitTransaction.order = shipmentInstance.returnOrder
}
```
If in the future somehow we want to allow multiple return orders per one shipment, then this won't work this way. 

Another approach would be to track it through the shipment, and simply find a shipment by incoming transaction/outgoing transacion and then look into its orders. But this seems to be awful to access it from template.

I think we can talk about this one on the Monday's Tech Huddle

Reply:
> If in the future somehow we want to allow multiple return orders per one shipment, then this won't work this way.

In that case, let's be more defensive. If you have more than one return order for a shipment, let's throw an UnsupportedOperationException.

Reply:
> Another approach would be to track it through the shipment, and simply find a shipment by incoming transaction/outgoing transacion and then look into its orders. But this seems to be awful to access it from template.

This feels safer to me but we can discuss on the tech huddle.